Community/Expire/automation/jsr223/python/community/expire/expire.py

```diff
@@ -103,127 +106,239 @@ def get_config(item_name):
+                        .plusMillis(int(cfg["time"].microseconds/1000)))
+    if (timers[item_name] is not None
+            and not timers[item_name].hasTerminated()):
+        timers[item_name].reschedule(t)
+    else:
+        timers[item_name] = ScriptExecution.createTimer(
+            t,
+            lambda: expired(ir.getItem(item_name),
+                            cfg["type"],
+                            cfg["state"],
+                            expire.log)
+        )
+
+
+def expire_load(event):
+
+    log = logging.getLogger("{}.Expire Load".format(LOG_PREFIX))
+    log.debug("Expire loading...")
+
+    # Keep track of items configured this pass
+    new_items = []
+
+    # Scan for items with valid expire config
+    for item_name in items:
+        cfg = get_config(item_name, log)
+        if cfg:
+            new_items.append(item_name)
+            timers[item_name] = timers.get(item_name, None) # don't clobber
+            log.debug("Expire configured for '{}' with timeout {} to {} '{}'"
+                      .format(item_name, cfg["time"],
+                      cfg["type"], cfg["state"]))
+
+    # Remove existing rule
+    if hasattr(expire, "UID"):
+        rules.remove(expire.UID)
+        delattr(expire, "triggers")
+        delattr(expire, "UID")
+
+    # Generate triggers
+    for item_name in new_items:
+        when("Item {} received update".format(item_name))(expire)
+
+    # Create expire rule
+    if hasattr(expire, "triggers"):
+        rule(
+            "Expire",
+            description=("Simulates the Expire1 binding, updating or commanding an "
+                         "Item after a configured amount of time."),
+            tags=["expire"]
+        )(expire)
+        if hasattr(expire, "UID"):
+            log.info("Expire loaded successfully")
+        else:
+            log.error("Failed to create Expire rule")
+    else:
+        log.info("Expire found no configured items")
+
+    # Drop items that no longer exist or have expire config
+    for item_name in timers:
+        if item_name not in new_items:
+            if item_name in items:
+                log.debug("Removing item '{}' as it no longer has a valid expire config"
+                          .format(item_name))
+            else:
+                log.debug("Removing item '{}' as it no longer exists"
+                          .format(item_name))
+            if (timers[item_name] is not None
+                    and not timers[item_name].hasTerminated()):
+                timers[item_name].cancel()
+            timers.pop(item_name, None)
+
+
+@log_traceback
+def scriptLoaded(*args):
+    """
+    Called at script load, sets up the reload rule.
+    """
+    log = logging.getLogger("{}.Expire Init".format(LOG_PREFIX))
+
+    import configuration
+    if hasattr(configuration, "expire_reload_item"):
+        item_name = configuration.expire_reload_item
+    else:
+        item_name = "expire_reload"
```

Comment:
The standard convention for Item names is they should start with a capital letter. I think that is a reasonable convention. I'd use ExpireReload or Expire_Reload to follow that convention for default Items.

Reply:
Personally I disagree with the naming patterns used throughout the forum and documentation for oh, so this is what I picked for a name. I knew this would probably be changed before merging since my naming pattern is a little different. It has also occurred to me that maybe we shouldn't have a default item name for it. Thoughts?

Reply:
Whether or not you agree with the convention, the convention exists and we should stick to it for libraries. 

I'm OK with not having a default though. The user needs to create the Item one way or the other. It's simpler to document and the code is a little simpler if we just force the user to supply the configuration variable and if they fail to do so disable the reload feature. I don't think having the default buys us much.

Reply:
I agree about the convention for libraries and not providing a default item name.